### PR TITLE
Updated systemjs-builder dependency to use jspm 0.17-beta instead

### PR DESCRIPTION
**Pull request by dennispg**
I thought I would make an update to your code to be compatible with the 0.17-beta release of jspm. Seems to work great for me!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jspm-tsc-update",
-  "version": "0.1.2",
+  "version": "0.2.0-beta",
```
Comment (fabiandev) [inline package.json]: Can you please revert the version number? I will then bump it alongside releasing a new version. Thanks!

**Review by fabiandev — CHANGES_REQUESTED** [review]

**Comment by fabiandev** [comment]
Thanks for this @dennispg! I've just requested a minor change.

**Comment by dennispg** [comment]
So I was thinking that this likely has breaking changes from your original version, because it had a dependency on `systemjs-builder 0.16.9` before and this branch is now switched over to use `jspm 0.17` instead.

I know you had mentioned you were using this in a live project, so I thought maybe having `0.2.0-beta` as a separate branch might be more wise? Like the same way how `jspm 0.17` is being maintained right now, where you have to do `npm install jspm@beta`.. I thought could do the same thing here.

But its up to you! I'm happy to still make the change as you requested if you would rather. Just let me know!

**Comment by fabiandev** [comment]
Sorry, I totally missed to reply. You are right, but even though the internal dependency changed, the behavior of this package should be the same, which is why I would tend to go straight to `0.2.0`. However, I also see your point of maintaining a separate branch until `jspm 0.17` is officially released. Another possibility would be to stick with `systemjs-builder 0.16` until `jspm 0.17` is out of beta. What do you think?

**Comment by dennispg** [comment]
I think the best measure may be in the project where you are currently using this. It would be a good reflection of any issues anyone else who might have installed this package might face. If it works as is, then I think going to straight `0.2.0` would be fine. Although maybe there is risk of unintended consequences that someone else may see that we wont?

For my use case, I rely on `jspm 0.17` so I would recommend against just sticking with `systemjs-builder 0.16` till `jspm 0.17` comes out of beta.

I think I am leaning towards maintaining a beta branch, but I will defer to you.

**Comment by fabiandev** [comment]
I will have a look at the behavior with the project where I'm using this package, however I think you're right and maintaining a beta/next branch might be a good idea. As there's still some things I'd like to implement (i.e., CLI options, and general improvements) I think I'd then tend to only introduce them to the separate branch we are talking about.

**Comment by dennispg** [comment]
I made a slight change to separate out the CLI functionality, just to enable my need to be able to integrate this into my gulp tasks.

I also made another more major change.. I removed the `pathmap.json` file you were creating. I believe it's purpose was to track any manual path entries a user might include in `tsconfig.json` from getting overwritten.

What I've done now instead, is basically to write the paths out to a file called `tsconfig.jspm.json` instead and then set the `extends` option in `tsconfig.json` to point to this standalone file.

Now you can make changes to the `paths` options in `tsconfig.json` without it being overwritten, which I think accomplishes the same function as what `pathmap.json` provided.

It also makes source control management of the `tsconfig.json` file a bit cleaner I think.

What do you think?

**Comment by fabiandev** [comment]
This looks very good, thanks! I'll try to merge your changes today.

**Comment by fabiandev** [comment]
@dennispg I'm sorry I couldn't make it yet. Unfortunately I'm away for at least a week, but this is definitely on my list!

**Review by fabiandev — APPROVED** [review]

**Pull request merged**